### DISCUSSION TITLE
Direct resource embeddings similar to Postgrest

### DISCUSSION THREAD

**Discussion by ishanray** [pocketbase/pocketbase#2371] [Polls]
Right now the expand column is used to get any relationship data.

In Postgrest the relation can be directly embedded in the result as so: 
`GET /films?select=title,directors(id,last_name)`

with the result being:
```
[
  { "title": "Workers Leaving The Lumière Factory In Lyon",
    "directors": {
      "id": 2,
      "last_name": "Lumière"
    }
  }
]
```

with the data model being:
![image](https://user-images.githubusercontent.com/5403397/234926079-aea5e8aa-b026-4f84-a3f8-515fa74310b1.png)

https://postgrest.org/en/stable/api.html#resource-embedding

**Comment by ganigeorgiev** [comment]
In my opinion it is better the expanded relations to be kept under a different field (aka. in our case under the `expand` key), so that you always have consistent type representation of the regular record fields (_eg. one benefit from this is that you can update a record with the same json format as the retrieve call_).

At least for v1, I'm not planning changing the above behavior.

**Reply by ishanray** [reply]
Makes sense. I was just wondering why it was under `expand`.

> one benefit from this is that you can update a record with the same json format as the retrieve call

Can you explain this a bit further. Thank you.

**Reply by ganigeorgiev** [reply]
Having consistent field types representation allows you to do something like:

```js
// fetch
let record = await pb.collection("example").getOne("RECORD_ID");

record.someField = "123";

// update directly with the record object
record = await pb.collection("example").update(record.id, record);
```
